be/src/olap/tablet_schema.cpp

```diff
@@ -902,14 +902,13 @@ void TabletColumn::append_sparse_column(TabletColumn column) {
     _num_sparse_columns++;
 }
 
-void TabletSchema::append_index(TabletIndex index) {
+void TabletSchema::append_index(TabletIndex&& index) {
     _indexes.push_back(std::move(index));
 }
 
 void TabletSchema::update_index(const TabletColumn& col, TabletIndex index) {
-    int32_t col_unique_id = col.unique_id();
-    const std::string& suffix_path =
-            col.has_path_info() ? escape_for_path_name(col.path_info_ptr()->get_path()) : "";
+    int32_t col_unique_id = col.is_extracted_column() ? col.parent_unique_id() : col.unique_id();
```

Comment:
Do you mean fixing a old bug here?  @eldenmoon can you have a look?

Reply:
yes, I will fix this error in branch-2.1

Reply:
DO NOT try to fix a bug in a refactor PR, fix it in another PR.

Reply:
Okay

Reply:
i think it's ok to modify this logic, since, update_index may not be called for variant type before